qcelemental/molparse/to_string.py

```diff
@@ -143,6 +144,21 @@ def to_string(molrec: Dict,
             'coordinates': 'cartesian',
         }
 
+    elif dtype == 'molpro':
+
+        atom_format = '{elem}'
+        ghost_format = 'Q'
```

Comment:
sure that's not a dummy? I found different instructions for ghost in my old notes (below). May as well construct the charge, mult, dummy commands here, I think.

```

# setup some QC program differences
if ( ($method eq "xyz") || ($method eq "qchem") || ($method eq "xdm") ) {

   $gh  = "Gh";   # to ghost an atom, replace its symbol with this and specify the atom type in the basis list
   $dm  = "X ";   # dummy atoms stored this way in resource file zmat
   $eql = "=";    # separate variable name and value in zmats
}
elsif ($method eq "molpro") {

   $gh  = "NaN";  # to ghost an atom, leave it alone in the geometry spec but feed its number to the dummy card
   $dm  = "Q ";   # can be Q or X, but the former can't be confused with symmetry specs
   $eql = "=";
   $memory = $memory/8.0;  # translate megabytes to megawords for molpro
}
elsif ( ($method eq "psi4") || ($method eq "sapt") || ($method eq "database") ) {

   $gh  = "NaN";  # to ghost an atom, leave it alone in the geometry and pick out the submolecules
   $dm  = "X ";   # looks to be X from the sample files
   $eql = "=";
}
elsif ($method eq "nwchem") {

   $gh  = "Bq";   # to ghost an atom, start it with Bq and specify a basis set for example, BqH or BqO
   $dm  = "X ";   # can be Bq or X, musn't have basis functions attached HOWEVER, doesn't actually like Bq with zmats
   $eql = " ";
}
elsif ($method eq "gaussian") {

   $gh  = "Bq";
   $dm  = "X ";
   $eql = " ";
}
```

Reply:
Hmm I suppose I need clarification on dummy atoms vs ghost atoms. It appears if you use Q or X in a Z matrix format it makes a center with no charge or basis functions (https://www.molpro.net/develop/master/doc/manual/node103.html). Otherwise if you want the atom to have basis functions but no charge you need to call the dummy card as you mention in your notes. 

So which should the ghost atom in QCElemental case refer to?

Reply:
The broadly used definitions are as below. molpro's "dummy" list is badly named.
* atom -- yes charge, yes basis fns
* ghost -- no charge, yes basis fns
* dummy -- is a point, no charge, no basis fns

qcschema.molecule doesn't acknowledge zmat, so everything's in cartesian and atom & ghost are all we're concerned with.

Reply:
Thanks for the clarification! I'll switch over `ghost_format = '{elem}'` and then add the appropriate dummy card for ghost atoms.

Reply:
Is there an easy way to accumulate the specified ghost atoms, so I can them write them to the dummy card?

Reply:
Molrec[‘real has the info you want. Like models.molecule.real, it’s a list of bools